MAGN-7450: SearchViewModel constructor should only take DynamoViewModel

aosyatnik opened the pull request:
### Purpose

Link to YouTrack:
[MAGN-7450](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7450) SearchViewModel constructor should only take DynamoViewModel

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] The level of testing this PR includes is appropriate

### Reviewers
@Benglin 

### FYIs
@pboyer

Benglin commented on src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs:
```diff
@@ -186,15 +186,28 @@ public ObservableCollection<NodeCategoryViewModel> BrowserRootCategories
+                return model;
+            }
+        }
+
```
Ouch, we shouldn't do this (i.e. shouldn't create `NodeSearchModel` internally which is disconnected from the rest of the system just for testing). Sorry I overlooked the part where unit-tests are still using the second constructor argument, in which case I am proposing the following:

```csharp
internal SearchViewModel(DynamoViewModel dynamoViewModel); // This is for main use case
internal SearchViewModel(NodeSearchModel model); // This is for test
```

Sorry about that!

Benglin commented:
Hi @aosyatnik, thanks for doing this. Sorry again that I overlooked the part where the constructor was used for unit-test cases that do not require `DynamoViewModel` but just `NodeSearchModel`. I hope my suggestion above makes sense.

Also another thing is, the [internal task ](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7450) includes another part about `RequestShowInCanvasSearch`, but you can make another PR for that part (so that I can merge this in soon after you're done tweaking the above). Thanks!

aosyatnik commented:
Hi, @Benglin , I've added new constructor. Also about second part of task. I've added it https://github.com/DynamoDS/Dynamo/pull/4543 long time ago :)

Benglin commented:
Thanks for fixing it, @aosyatnik. LGTM! Hi @ramramps, I'm going to be out of town for the week, once the conflict is resolved please help to merge it, thanks :)

aosyatnik commented:
Hi, @ramramps , updated this branch with the last master.
